### PR TITLE
Faster argmax/argmin kernels

### PR DESCRIPTION
**Pull request by jbobba**

### CODE REVIEW

File: src/ngraph/runtime/cpu/cpu_builder.hpp [hunk 1]
```diff
@@ -94,6 +94,24 @@
     else                                                                                           \
         throw ngraph_error("Unsupported rank " + std::to_string(R) + " for kernel " #K);
 
+#define SELECT_RANK2(KV, IT, OT, R, K)                                                             \
+    if (R == 1)                                                                                    \
```
Comment (Krovatkin) [inline src/ngraph/runtime/cpu/cpu_builder.hpp]: nitpick: can't we use `case` here?

File: src/ngraph/runtime/cpu/cpu_builder.hpp [hunk 2]
```diff
@@ -94,6 +94,24 @@
...
+    else if (R == 3)                                                                               \
+        KV = K<IT, OT, 3>;                                                                         \
+    else if (R == 4)                                                                               \
+        KV = K<IT, OT, 4>;                                                                         \
+    else if (R == 5)                                                                               \
+        KV = K<IT, OT, 5>;                                                                         \
+    else if (R == 6)                                                                               \
+        KV = K<IT, OT, 6>;                                                                         \
+    else if (R == 7)                                                                               \
+        KV = K<IT, OT, 7>;                                                                         \
```
Comment (Krovatkin) [inline src/ngraph/runtime/cpu/cpu_builder.hpp]: out of curiosity; why do we stop at 7?

File: src/ngraph/runtime/cpu/kernel/argmax.hpp [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+                            void* output,
+                            const Shape& input_shape,
+                            const Shape& output_shape,
+                            size_t axis,
+                            int arena)
+                {
+                    Eigen::array<Eigen::Index, Rank - 1> out_dims;
+                    Eigen::array<Eigen::Index, Rank> in_dims;
+
+                    for (int i = 0; i < Rank; i++)
```
Comment (Krovatkin) [inline src/ngraph/runtime/cpu/kernel/argmax.hpp]: nitpick: Maybe we should consider conversion functions (`Shape` to `Eigen::array`)?

File: src/ngraph/runtime/cpu/cpu_builder.hpp (hunk 2 above)
Comment (aprocter) [reply src/ngraph/runtime/cpu/cpu_builder.hpp]: FWIW, TensorFlow seems to stop there for much of its stuff too. I don't really know why. I guess it's a lucky number?

File: src/ngraph/runtime/cpu/cpu_builder.hpp [hunk 4]
```diff
@@ -94,6 +94,24 @@
     else                                                                                           \
         throw ngraph_error("Unsupported rank " + std::to_string(R) + " for kernel " #K);
 
+#define SELECT_RANK2(KV, IT, OT, R, K)                                                             \
+    if (R == 1)                                                                                    \
+        KV = K<IT, OT, 1>;                                                                         \
```
Comment (rkimballn1) [inline src/ngraph/runtime/cpu/cpu_builder.hpp]: All code blocks get curly braces

File: src/ngraph/runtime/cpu/kernel/argmax.hpp (hunk 3 above)
Comment (ayzhuang) [reply src/ngraph/runtime/cpu/kernel/argmax.hpp]: utility function?

**Pull request merged**